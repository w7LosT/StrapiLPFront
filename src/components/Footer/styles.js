import styled, { css } from "styled-components";
import {Text} from '../TextComponent/styles';
import {Container as SectionComponent} from '../SectionComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    border-top: 1px solid rgba(0, 0, 0, .3);

    & a{
      color: inherit;
      text-decoration: none;
    }

    & ${Text}{
      font-size: ${theme.fonts.sizes.sm};
    }
  `}
`;
