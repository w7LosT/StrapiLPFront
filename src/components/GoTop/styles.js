import styled, { css } from "styled-components";

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    bottom: 30px;
    right: 15px;
    width: 50px;
    height: 50px;
    font-size: ${theme.fonts.sizes.lg};
    color: ${theme.default.darkBG};
    background-color: ${theme.default.colorWhite};
    border-radius: 100%;
    border: 2px solid ${theme.default.darkBG};
    box-shadow: 0 3px 6px rgba(0, 0, 0, .4);
    z-index: 6;
    opacity: .7;
    transition: all .3s ease-in-out;

    &:hover{
      opacity: 1;
    }
  `}
`;
