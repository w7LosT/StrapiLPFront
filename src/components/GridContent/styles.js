import styled, { css } from "styled-components";

export const Container = styled.div`
  ${({ theme, background }) => css`
    max-width: 580px;
    margin: 0 auto;
    text-align: center;

    & h1{
      background: none;
      color: ${ background ? theme.default.colorWhite : theme.default.primaryColor };
    }

    & a{
      color: ${ background ? theme.default.colorWhite : theme.default.primaryColor };
    }
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.lg} 0;
  `}
`;
