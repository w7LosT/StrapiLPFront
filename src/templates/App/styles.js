import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${(props) => css`
    background: purple;
    color: ${props.color};
  `}
`;
