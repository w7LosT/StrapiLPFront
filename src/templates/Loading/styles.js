import styled, { css, keyframes } from "styled-components";

const rotate = () => keyframes`
  0% {
    transform: translate(-50%, -50%) rotate(0deg);
  }

  100%{
    transform: translate(-50%, -50%) rotate(360deg);
  }
`;

export const Container = styled.div`
  ${({ theme }) => css`
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    z-index: 10;
    background-color: rgba(0, 0, 0, .7);

    &:before, &:after{
      content: '';
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      border-width: .5rem;
      border-color: transparent;
      border-style: solid;
      border-radius:  50%;
    }

    &:after{
      width: 6rem;
      height: 6rem;
      border-left: .5rem solid #fff;
      border-top: .5rem solid #fff;
      animation: ${rotate} 1s linear infinite;
    }

    &:before{
      width: 3rem;
      height: 3rem;
      border-left: .5rem solid #fff;
      border-top: .5rem solid #fff;
      animation: ${rotate} .5s linear reverse infinite;
    }
  `}
`;
