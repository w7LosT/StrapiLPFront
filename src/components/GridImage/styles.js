import styled, { css } from "styled-components";
import { Title as  HeadingContainer } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    padding: 10px;
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.lg};
    padding: ${theme.spacings.xlg} 0;
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    transition: all .3s ease;
    box-shadow: 0 3px 6px rgba(0, 0, 0, .5);

    &:hover{
      transform: scale(1.1);
      box-shadow: 0 6px 8px rgba(0, 0, 0, .5);

    }
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;

  `}
`;
