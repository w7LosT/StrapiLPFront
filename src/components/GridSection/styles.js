import styled, { css } from "styled-components";
import { Title as  HeadingContainer } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    max-width: 1140px;
    margin: 0 auto;
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.lg};
    padding: ${theme.spacings.xlg} 0;
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer}{
      position: relative;
      left: 35px ;
      margin-bottom: 10px;
    }

    ${HeadingContainer}::before{
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.fonts.sizes.xlg};
      top: -1rem;
      left: -3rem;
      transform: rotate(5deg);
    }
  `}
`;
