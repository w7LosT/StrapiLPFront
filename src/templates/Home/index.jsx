import * as Styled from './styles';
import { Heading } from '../../components/Heading';
import { Text } from '../../components/TextComponent';

function Home() {
  return (
    <div className="App">
      <Styled.Wrapper>
        <Heading>
          Oi
        </Heading>
      </Styled.Wrapper>
    </div>
  );
}

export default Home;
