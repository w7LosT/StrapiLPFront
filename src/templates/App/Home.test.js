import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/theme';
import Home from './index';

export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};

test('renders learn react link', () => {
  const {debug} = renderTheme(<Home />);
  // debug();
  const headingContainer = screen.getByRole('heading', {name: "Oi"}).parentElement;
  expect(headingContainer).toHaveStyle({
    "background-color": theme.default.mainBG,
  });
  expect(headingContainer).toMatchSnapshot();
});
