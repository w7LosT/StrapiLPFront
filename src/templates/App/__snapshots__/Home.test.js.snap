// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  class="sc-bdvvtL dbZhSS"
  color="green"
>
  <h1>
    Oi
  </h1>
</div>
`;
